src/GoogleFitOnFhir/Clients/GoogleFit/Handlers/GoogleFitHandler.cs

```diff
@@ -0,0 +1,74 @@
+﻿// -------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+using System;
+using System.Threading.Tasks;
+using EnsureThat;
+using GoogleFitOnFhir.Clients.GoogleFit.Responses;
+using GoogleFitOnFhir.Common;
+using GoogleFitOnFhir.Services;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.Extensions.Logging;
+using Microsoft.Health.Common.Handler;
+
+namespace GoogleFitOnFhir.Clients.GoogleFit.Handlers
+{
+    public class GoogleFitHandler : IResponsibilityHandler<RoutingRequest, Task<IActionResult>>
+    {
+        private const string GoogleFitAuthorizeRequest = "api/googlefit/authorize";
+        private const string GoogleFitCallbackRequest = "api/googlefit/callback";
+        private IAuthService _authService;
+        private IUsersService _usersService;
+        private readonly ILogger<GoogleFitHandler> _logger;
+
+        public GoogleFitHandler(IAuthService authService, IUsersService usersService, ILogger<GoogleFitHandler> logger)
+        {
+            _authService = authService;
+            _usersService = usersService;
+            _logger = logger;
+        }
+
+        public Task<IActionResult> Evaluate(RoutingRequest request)
+        {
+            var path = EnsureArg.IsNotNullOrWhiteSpace(request.HttpRequest.Path.Value?[1..]);
```

Comment:
I think the problem with this implementation is that if this value is null or whitespace, this throws and the other handlers don't have an opportunity to process the RoutingRequest. I think we need to handle this case gracefully and try and prevent the possibility of exceptions from happening in the first place, or catch and log exceptions.